src/core/dense_set.cc

```diff
@@ -229,11 +229,13 @@ auto DenseSet::FindEmptyAround(uint32_t bid) -> ChainVectorIterator {
 }
 
 void DenseSet::Reserve(size_t sz) {
-  sz = std::min<size_t>(sz, kMinSize);
+  sz = std::max<size_t>(sz, kMinSize);
 
   sz = absl::bit_ceil(sz);
-  capacity_log_ = absl::bit_width(sz);
-  entries_.reserve(sz);
+  if (sz > entries_.size()) {
+    entries_.resize(sz);
```

Comment:
I was not sure about this so I looked on `dense_set.cc` to make sure that we don't call `push_back` anywhere (because otherwise this resize would go to waste, since push_back would grow beyond that) but we are doing only raw accesses so this should be fine